src/components/cylc/tree/deltas.js

```diff
@@ -181,71 +176,76 @@ function handleInitialDataBurst (deltas, tree) {
-      //      which we can use to create the tree structure.
-      //   2. Or this means that after the shutdown (when we delete the tree), we received a delta.
-      //      In this case we don't really have any way to fix the tree.
-      // In both cases, actually, the user has little that s/he could do, besides refreshing the
-      // page. So we fail silently and wait for a request with the initial data.
-      if (!deltas.added || !deltas.added.workflow) {
-        // eslint-disable-next-line no-console
-        console.error('Received a delta before the workflow initial data burst')
-        store.dispatch('setAlert', new AlertModel('Received a delta before the workflow initial data burst. Please reload your browser tab to retrieve the full flow state', null, 'error'))
-        return
-      }
-      try {
-        handleInitialDataBurst(deltas, tree)
-      } catch (error) {
-        // eslint-disable-next-line no-console
-        console.error('Error applying initial data burst for deltas', error, deltas)
-        store.dispatch('setAlert', new AlertModel('Error applying initial data burst for deltas. Please reload your browser tab to retrieve the full flow state', null, 'error'))
-        throw error
-      }
-    } else {
-      // the tree was created, and now the next messages should contain
-      // 1. new data added under deltas.added (but not in deltas.added.workflow)
-      // 2. data updated in deltas.updated
-      // 3. data pruned in deltas.pruned
-      try {
-        handleDeltas(deltas, tree)
-      } catch (error) {
-        // eslint-disable-next-line no-console
-        console.error('Unexpected error applying deltas', error, deltas)
-        throw error
+  }
+  // Safe check in case the tree is empty.
+  if (CylcTree.isEmpty(workflow)) {
+    // When the tree is empty, we have two possible scenarios:
+    //   1. This means that we will receive our initial data burst in deltas.added.workflow
+    //      which we can use to create the tree structure.
+    //   2. Or this means that after the shutdown (when we delete the tree), we received a delta.
+    //      In this case we don't really have any way to fix the tree.
+    // In both cases, actually, the user has little that s/he could do, besides refreshing the
+    // page. So we fail silently and wait for a request with the initial data.
+    if (!deltas.added || !deltas.added.workflow) {
+      return {
+        errors: [
+          ['Received a delta before the workflow initial data burst']
+        ]
       }
     }
-  } else {
-    throw Error('Workflow tree subscription did not return data.deltas')
+  }
+  // the tree was created, and now the next messages should contain
+  // 1. new data added under deltas.added (but not in deltas.added.workflow)
+  // 2. data updated in deltas.updated
+  // 3. data pruned in deltas.pruned
+  try {
+    return handleDeltas(deltas, workflow, lookup, options)
```

Comment:
**Note**: We are handling only deltas, no more handling initial data burst. I have modified the queries to use a flat structure, and whether it's the initial burst or not, makes no difference to the UI. I simply add every data returned to a dictionary and store in Vuex. Other Vuex actions (e.g. `applyTreeDeltas`) can then access the dictionary and use it to retrieve created values from delta, avoiding this way creating new - and duplicated - data.